golang/cosmos/x/swingset/types/params.go

```diff
@@ -165,76 +171,80 @@ func validateQueueMax(i interface{}) error {
 	return nil
 }
 
+func validateVatCleanupBudget(i interface{}) error {
+	entries, ok := i.([]UintMapEntry)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	hasDefault := false
+	for _, entry := range entries {
+		if entry.Key == VatCleanupDefault {
+			hasDefault = true
+			break
+		}
+	}
+	if len(entries) > 0 && !hasDefault {
+		return fmt.Errorf("`default` must be present in a non-empty vat cleanup budget")
+	}
+	return nil
+}
+
 // UpdateParams appends any missing params, configuring them to their defaults,
 // then returning the updated params or an error. Existing params are not
 // modified, regardless of their value, and they are not removed if they no
 // longer appear in the defaults.
 func UpdateParams(params Params) (Params, error) {
-	newBpu, err := appendMissingDefaultBeansPerUnit(params.BeansPerUnit, DefaultBeansPerUnit())
+	newBpu, err := appendMissingDefaults(params.BeansPerUnit, DefaultBeansPerUnit())
 	if err != nil {
 		return params, err
 	}
-	newPff, err := appendMissingDefaultPowerFlagFees(params.PowerFlagFees, DefaultPowerFlagFees)
+	newPff, err := appendMissingDefaults(params.PowerFlagFees, DefaultPowerFlagFees)
 	if err != nil {
 		return params, err
 	}
-	newQm, err := appendMissingDefaultQueueSize(params.QueueMax, DefaultQueueMax)
+	newQm, err := appendMissingDefaults(params.QueueMax, DefaultQueueMax)
+	if err != nil {
+		return params, err
+	}
+	newVcb, err := appendMissingDefaults(params.VatCleanupBudget, DefaultVatCleanupBudget)
```

Comment:
ok so *now* my current beliefs are:
* we get one transaction type to influence all swingset-related parameters (four sets: beansPerUnit, powerFlagFees, queueMax, vatCleanupBudget)
* if any field of any set is not included in the transaction/proposal, and *is* included in e.g. `DefaultBeansPerUnit` or `DefaultPowerFlagFees` or `DefaultQueueMax` or `DefaultVatCleanupBudget`, then that field is populated here
* so if someone makes a governance parameter proposal to change something in `params.QueueMax`, and they forget to look at the chain and copy in the current fields from `VatCleanupBudget`, they'll be accidentally resetting the cleanup budget back to the default
* also, because both `.default` and `.kv` are in `DefaultVatCleanupBudget`, there is no way to say "please stop overriding `kv` and just let it use `.default` like everything else does"
  * the closest you could achieve is by saying `{ default: 5, kv: 5 }`

That works, it's weird but I suspect it's the best we can get from a one-method-per-module parameter-setting scheme.